tests/debug.rs

```diff
@@ -1360,7 +1441,7 @@ mod generic {
     }
 
     #[derive(Debug)]
-    struct AliasedFieldUnnamedGenericStruct<T>(#[debug("{_0}", _0 = _1)] T, i32);
+    struct AliasedFieldUnnamedGenericStruct<T>(#[debug("{_2}", _2 = _1)] T, i32);
```

Comment:
@JelteF I don't really like the necessity of this limitation. This is contra-intuitive for me and doesn't feel smooth from the user experience.

How about we apply the `_0 = *_0` dereferencing only for:
1. arguments requiring `Pointer`-formatting;
2. and the ones not using a defined named argument already?

So, this concrete example will expand as before. And something complex like
```rust
#[debug("{_0:p} + {_1:p} = {_0}", _0 = *_1)]
```
will expand to
```rust
write!(f, "{_0:p} + {_1:p} = {_0}", _0 = *_1, _1 = *1)
```

Reply:
If you're short on time with this, I'll do it, if you don't mind.

Reply:
I don't really see a reason to only do this for Pointer formatting only. So I'd prefer to do this change consistently.

Regarding allowing people to override the default bindings by using their name (e.g. `_0 = *_1`), I understand that argument. But honestly I think it's quite confusing to do that anyway. I mean for me parsing what your example means takes much more effort than if you'd use a name that didn't already have another meaning.

If you feel strongly about this, I'm definitely willing to accept your second suggestion. But imho, the current restriction only impacts code that's already hard to read.

Reply:
@JelteF 

> If you feel strongly about this, I'm definitely willing to accept your second suggestion.

Yup, I think I'm quite strong about it. Can imagine perfectly well how this becomes a burden in some auto-generated code.

I did the second suggestion in the https://github.com/JelteF/derive_more/pull/381/commits/842ee72734122a113291ef7e7d75983f797d082b.